src/main/java/hu/blackbelt/structured/map/proxy/MapHolder.java

```diff
@@ -28,4 +28,5 @@ public interface MapHolder {
 
     Map<String, Object> getOriginalMap();
 
+    <T> T adaptTo(Class<T> clazz);
```

Comment:
`adaptTo` should have it's own interface? We are using this same method without a corresponding interface to implement.